storage/2020-08-04/queue/messages/update.go

```diff
@@ -26,90 +29,77 @@ type UpdateInput struct {
+		MessageText: input.Message,
+	})
 	if err != nil {
-		result = autorest.Response{Response: resp}
-		err = autorest.NewErrorWithError(err, "messages.Client", "Update", resp, "Failure sending request")
-		return
+		return resp, fmt.Errorf("marshalling request: %v", err)
 	}
 
-	result, err = client.UpdateResponder(resp)
+	body := xml.Header + string(marshalledMsg)
+	req.Body = io.NopCloser(bytes.NewReader([]byte(body)))
+	req.ContentLength = int64(len(body))
+	req.Header.Set("Content-Length", strconv.Itoa(len(body)))
+
+	resp.HttpResponse, err = req.Execute(ctx)
 	if err != nil {
-		err = autorest.NewErrorWithError(err, "messages.Client", "Update", resp, "Failure responding to request")
+		err = fmt.Errorf("executing request: %+v", err)
 		return
 	}
 
 	return
 }
 
-// UpdatePreparer prepares the Update request.
-func (client Client) UpdatePreparer(ctx context.Context, accountName, queueName string, messageID string, input UpdateInput) (*http.Request, error) {
-	pathParameters := map[string]interface{}{
-		"queueName": autorest.Encode("path", queueName),
-		"messageID": autorest.Encode("path", messageID),
-	}
-
-	queryParameters := map[string]interface{}{
-		"popreceipt":        autorest.Encode("query", input.PopReceipt),
-		"visibilitytimeout": autorest.Encode("query", input.VisibilityTimeout),
-	}
-
-	headers := map[string]interface{}{
-		"x-ms-version": APIVersion,
-	}
-
-	body := QueueMessage{
-		MessageText: input.Message,
-	}
-
-	preparer := autorest.CreatePreparer(
-		autorest.AsContentType("application/xml; charset=utf-8"),
-		autorest.AsPut(),
-		autorest.WithBaseURL(endpoints.GetQueueEndpoint(client.BaseURI, accountName)),
-		autorest.WithPathParameters("/{queueName}/messages/{messageID}", pathParameters),
-		autorest.WithQueryParameters(queryParameters),
-		autorest.WithXML(body),
-		autorest.WithHeaders(headers))
-	return preparer.Prepare((&http.Request{}).WithContext(ctx))
+type updateOptions struct {
+	input UpdateInput
 }
 
-// UpdateSender sends the Update request. The method will close the
-// http.Response Body if it receives an error.
-func (client Client) UpdateSender(req *http.Request) (*http.Response, error) {
-	return autorest.SendWithSender(client, req,
-		azure.DoRetryWithRegistration(client.Client))
+func (u updateOptions) ToHeaders() *client.Headers {
+	return nil
 }
 
-// UpdateResponder handles the response to the Update request. The method always
-// closes the http.Response Body.
-func (client Client) UpdateResponder(resp *http.Response) (result autorest.Response, err error) {
-	err = autorest.Respond(
-		resp,
-		client.ByInspecting(),
-		azure.WithErrorUnlessStatusCode(http.StatusNoContent),
-		autorest.ByClosing())
-	result = autorest.Response{Response: resp}
+func (u updateOptions) ToOData() *odata.Query {
+	return nil
+}
 
-	return
+func (u updateOptions) ToQuery() *client.QueryParams {
+	out := &client.QueryParams{}
+	out.Append("visibilitytimeout", strconv.Itoa(u.input.VisibilityTimeout))
```

Comment:
there's no validation for this one - does this want to be only set when it's `> 0`?

Reply:
I think we want to leave this as it is, `visibilitytimeout` is required in update and 0 is a valid value it seems: https://learn.microsoft.com/en-us/rest/api/storageservices/update-message#uri-parameters